tuf/api/metadata.py

```diff
@@ -320,7 +317,15 @@ def sign(
 
             signed_serializer = CanonicalJSONSerializer()
 
-        signature = signer.sign(signed_serializer.serialize(self.signed))
+        try:
+            signature = signer.sign(signed_serializer.serialize(self.signed))
+        except (
+            sslib_exceptions.CryptoError,
+            sslib_exceptions.UnsupportedAlgorithmError,
```

Comment:
This `except` block assumes that `signer` is an `SSlibSigner`, but the caller could pass any implementation of the `Signer`  interface. Ideally the `Signer` interface would define what exceptions to expect here, like storage or serialization interfaces do (maybe this is worth a ticket in securesystemslib?).

Should we just except all and re-raise as UnsignedMetadataError?

Reply:
good catch. For this PR (and 1.0) the last suggestion sounds reasonable to me.

Reply:
Really great catch.
Totatlly forgot that `signed_serializer` is an object implementing the general `securesystemslib.signer.Signer` interface.

Reply:
> Ideally the Signer interface would define what exceptions to expect here, like storage or serialization interfaces do (maybe this is worth a ticket in securesystemslib?).


The `securesystemslib` issue about this: https://github.com/secure-systems-lab/securesystemslib/issues/386